std/container/array.d

```diff
@@ -349,7 +356,7 @@ if (!is(Unqual!T == bool))
             import core.checkedint : mulu;
             bool overflow;
             const sz = mulu(elements, T.sizeof, overflow);
-            if (overflow) assert(0);
+            assert(!overflow, "Trying to allocate more than size_t.max bytes.");
```

Comment:
Ok, there is `HLT` indeed. That makes things clear to me, thank you very much! I'm gonna to revert the previous code here then.